src/redisearch.h

```diff
@@ -27,6 +27,11 @@ typedef uint16_t t_fieldIndex;
 struct timespec;
 typedef struct timespec t_expirationTimePoint;
 
+typedef uint64_t t_uniqueId;
+#define SIGN_CHAR_LENGTH 0 // t_uniqueId is unsigned
+#define LOG_10_ON_256_UPPER_BOUND 3 // 2^8 = 10 ^ y, 2^16 = 2^8 * 2^8 = 10^y * 10^y = 10^2y -> y == 2.40824 -> upper bound for y is 3
+#define MAX_UNIQUE_ID_TEXT_LENGTH_UPPER_BOUND ((sizeof(t_uniqueId) * LOG_10_ON_256_UPPER_BOUND) + SIGN_CHAR_LENGTH)
```

Comment:
Why do you want to compute `MAX_UNIQUE_ID_TEXT_LENGTH_UPPER_BOUND`, it is `strlen(UINT64_MAX) = 20`, it isn't?

Reply:
Yes, but Omer didn't like the fact it was independent from the type of t_uniqueId.
So this is the compromise we reached.